room/compiler-processing/src/main/java/androidx/room/compiler/processing/javac/JavacProcessingEnv.kt

```diff
@@ -183,6 +185,46 @@ internal class JavacProcessingEnv(
         } as T
     }
 
+    // TODO: Add a cache layer
```

Comment:
we can skip these caches. particularly, these get cached in the TypeElement anyways and it is very unlikely to find an executable element without a type element reference first.